### ISSUE TITLE
ValueError when using NumPy arrays as rows in Matrix constructor in SymPy 1.5

### ISSUE THREAD

**Issue by jan-mue** [sympy/sympy#18268]
Version 1.5 of the package breaks some of my code. When an iterable of NumPy arrays is used as argument to the sympy.Matrix constructor, the following exception is raised:

```
Traceback (most recent call last):
  File "<input>", line 6, in <module>
  File "/Users/jan/.local/share/virtualenvs/geometer-zell7LBw/lib/python3.7/site-packages/sympy/matrices/dense.py", line 430, in __new__
    return cls._new(*args, **kwargs)
  File "/Users/jan/.local/share/virtualenvs/geometer-zell7LBw/lib/python3.7/site-packages/sympy/matrices/dense.py", line 442, in _new
    rows, cols, flat_list = cls._handle_creation_inputs(*args, **kwargs)
  File "/Users/jan/.local/share/virtualenvs/geometer-zell7LBw/lib/python3.7/site-packages/sympy/matrices/matrices.py", line 2623, in _handle_creation_inputs
    if not row:
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

The following code works in SymPy 1.4 but not in 1.5:

```python
import sympy
import numpy as np

row1 = np.array([1, 2])
row2 = np.array([3, 4])
sympy.Matrix([row1, row2])
```

**Comment by ethankward** [comment]
Bisected to commit e05170e915f2fe4813ffcc357487cb5f6ecedee8.

**Comment by oscarbenjamin** [comment]
I'm adding the 1.6 milestone because this was a regression in 1.5 that should be fixed before 1.6.
